### ISSUE TITLE
TypeError in Zod Library When Calling getPool Function in Uniswap Factory Contract (web3.js 4.x)

### ISSUE THREAD

**Issue by CoinHunter314159** [ethereum/web3.js#6376]
#### Expected behavior
When calling the getPool function of the Uniswap factory contract on the Arbitrum Goerli network, I expect to receive the pool address.
#### Actual behavior
When attempting to call the getPool function, I encountered a TypeError related to the Zod library:
`node_modules/zod/lib/types.js:199
const result = check(val);
               ^
TypeError: check is not a function
`
#### Steps to reproduce the behavior
1.Connect to the Arbitrum Goerli network using web3.js.
2.Define the ABI and address for the Uniswap factory contract.
3.Create a contract instance and call the getPool function.
4.Observe the TypeError in the console.

`
const { Web3, Contract, } = require('web3');

// Connect to Ethereum node
const web3 = new Web3(new Web3.providers.HttpProvider('')); //hide the RPC

// Define Contract ABI and Address
const contractABI = [{"inputs":[],"stateMutability":"nonpayable","type":"constructor"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"uint24","name":"fee","type":"uint24"},{"indexed":true,"internalType":"int24","name":"tickSpacing","type":"int24"}],"name":"FeeAmountEnabled","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"oldOwner","type":"address"},{"indexed":true,"internalType":"address","name":"newOwner","type":"address"}],"name":"OwnerChanged","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"token0","type":"address"},{"indexed":true,"internalType":"address","name":"token1","type":"address"},{"indexed":true,"internalType":"uint24","name":"fee","type":"uint24"},{"indexed":false,"internalType":"int24","name":"tickSpacing","type":"int24"},{"indexed":false,"internalType":"address","name":"pool","type":"address"}],"name":"PoolCreated","type":"event"},{"inputs":[{"internalType":"address","name":"tokenA","type":"address"},{"internalType":"address","name":"tokenB","type":"address"},{"internalType":"uint24","name":"fee","type":"uint24"}],"name":"createPool","outputs":[{"internalType":"address","name":"pool","type":"address"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint24","name":"fee","type":"uint24"},{"internalType":"int24","name":"tickSpacing","type":"int24"}],"name":"enableFeeAmount","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint24","name":"","type":"uint24"}],"name":"feeAmountTickSpacing","outputs":[{"internalType":"int24","name":"","type":"int24"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"","type":"address"},{"internalType":"address","name":"","type":"address"},{"internalType":"uint24","name":"","type":"uint24"}],"name":"getPool","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"owner","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"parameters","outputs":[{"internalType":"address","name":"factory","type":"address"},{"internalType":"address","name":"token0","type":"address"},{"internalType":"address","name":"token1","type":"address"},{"internalType":"uint24","name":"fee","type":"uint24"},{"internalType":"int24","name":"tickSpacing","type":"int24"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"_owner","type":"address"}],"name":"setOwner","outputs":[],"stateMutability":"nonpayable","type":"function"}];
const contractAddress = '0x4893376342d5D7b3e31d4184c08b265e5aB2A3f6';

// Create Contract Instance

const contract = new Contract(
        contractABI,
        '0x4893376342d5D7b3e31d4184c08b265e5aB2A3f6',
        web3,
    );

const run = async () => {
    const result = await contract.methods.getPool("0x091E8F335C77cc8b76645Ec2f3ADCaB1AaD82876", "0x1ba16Ad67758e3ec512C7C961794970105e3c192", 3000).call();
};

run();
`

#### Environment

npm version: 9.5.1
Node version: 18.16.0
web3.js version: 4.0.3, 4.1.1 (Issue persists in both versions)
OS: MAC OS 13.4.1 (c) (22F770820d)
Contract Address: 0x4893376342d5D7b3e31d4184c08b265e5aB2A3f6

**Comment by avkos** [comment]
Thanks for your contribution. I've reproduced this locally. the problem is in type uint24. 
now we support only these int/unit formatters:
```js 
for (let i = 3; i <= 8; i += 1) {
const bitSize = 2 ** i;
`uint${bitSize}`
`int${bitSize}`
}
```

 you can change it to a `uint` to make it works right now but definitely we need to fix this problem.

**Comment by mconnelly8** [comment]
Fixed with #6434
